core/network/nic.go

```diff
@@ -211,21 +211,21 @@ func (i *InterfaceInfo) IsVLAN() bool {
 }
 
 // CIDRAddress returns Address.Value combined with CIDR mask.
-func (i *InterfaceInfo) CIDRAddress() string {
+func (i *InterfaceInfo) CIDRAddress() (string, error) {
 	primaryAddr := i.PrimaryAddress()
 	if i.CIDR == "" || primaryAddr.Value == "" {
-		return ""
+		return "", nil
```

Comment:
Why not return an `errors.NotValid` here? It feels strange to have a method returning an empty value when it encounters errors and an `"", nil` here.

Reply:
I returned a `NotFound`, which I think is more appropriate given the usage context for cloud-init generation. In this case, we fall back to DHCP for the device.